### DISCUSSION TITLE
Inference on custom trained model

### DISCUSSION THREAD

**Discussion by kaphleamrit2** [obss/sahi#672] [Q&A] [answered]
I am trying to run object detection and instance segmentation on large images using SAHI. I have trained MaskRCNN model on Detectron2 and save model as "/content/output/object_detection/model_final.pth" and config file as "/content/OD_cfg.pkl". 
Then I used 

```
detection_model = Detectron2DetectionModel(
    model_path="/content/output/object_detection/model_final.pth",
    config_path="/content/OD_cfg.pkl",
    confidence_threshold=0.5,
    image_size=640,
    device="cpu", # or 'cuda:0'
)
```
But I am getting the error: /content/OD_cfg.pkl not available in Model Zoo!. I can't run the prediction. I think I am missing something here. Does anybody have any idea what's going on here? How can I load custom trained model and config file? Please help me.
![Screenshot 2022-03-08 104127](https://user-images.githubusercontent.com/58353307/157284018-764908b1-93ce-4f6e-9aec-2a4106fdb12d.png)

**Comment by fcakyon** [comment, marked as answer]
Thanks a lot for the issue @kaphleamrit2 !

We currently only support yaml config files for Detectron2 models. Check this post to details on how to save model config as yaml file: https://medium.com/codable/perform-sliced-inference-and-detailed-error-analysis-using-detectron2-models-37063e33073f

**Comment by kaphleamrit2** [comment]
Thank you @fcakyon. It works perfectly after I used yaml config file. Thanks for the help.

**Comment by fcakyon** [comment]
Glad to hear that @kaphleamrit2 🚀
